docs/proposals/1199-multitenant-api-proposal/README.md

```diff
@@ -0,0 +1,250 @@
+  - Traffic splitting models & modelName rewrite
+  - Criticality
+
+### Non-Goals
+- Addressing security concerns with the API, this is currently expected to either be:
+  - Entirely contained within a trusted system
+  - Or auth handled upstream
+- IGW implementing a custom auth mechanism
+
+
+## Definitions
+
+- **Tenant** (synonymous with: *Flow* or *Identity*) - In the context of Inference Gateway these names are synonymous. Kuberenetes chooses the term ***tenant*** as described [here](https://kubernetes.io/docs/concepts/security/multi-tenancy/#tenants).
+
+# Proposal
+
+Discussion of the problem(s) can be seen in the linked documents. Here we will describe the new API surface.
+
+## Phase 1
+
+### Naming
+This API solves 3 general pillars of problem, that can also be categorized into 2 areas:
+
+ - This API describes Resource Sharing (Criticality/Fairness)
+ - This API describes Identification (used in Fairness)
+ - This API describes Specific Request Policy (SLO)
+
+
+As such, the name of the API should convey these concepts well. The `Inference-` prefix will remain, as that is a succint & accepted term related to generative AI serving.
+
+The accompanying name should also convey some or all of the other pillars, some of the names that have been considered:
+
+- `Tenant`
+  - Pros:
+    - Resource Sharing is implicit due to evocation of the 'multitenancy' concept
+	- Identification is well understood from a term like Tenant
+  - Cons: 
+    - Due to the prevalance of the 'multitenancy' concept, the term of tenant may clash with a user-defined term of 'tenant' causing a confusing interaction
+	- Does not convey Specific Request Policy well
+
+- `Flow`
+  - Pros:
+    - A common networking term helpful in describing request traffic rate control 
+  - Cons:
+    - Doesnt well describe Identification or Resource Sharing
+
+- `Objectives` (preferred)
+  - Pros:
+    - Specific Request Policy is easily understood
+	- Resource Sharing is a natural fit
+  - Cons:
+    - Identity is not well conveyed
+
+
+### CRD spec
+
+This CRD definition is a slimmed version of InferenceModel with a name change. Example here:
+
+```golang
+type InferenceObjectives struct {
+  metav1.TypeMeta
+  metav1.ObjectMeta
+
+  Spec InferenceObjectivesSpec
+}
+
+type InferenceObjectivesSpec struct {
+  PoolRef InferenceObjectReference
+
+  Criticality *int
+}
+
+```
+
+### Other changes
+- The EPP will expose a flag to define the header key that will be used to assign InferenceObjectives to
+- The modelName rewrite functionality will be included into EPP as a core feature (also handled by header) 
+- Traffic splitting on model name will be pushed to HTTPRoute using the BBR tool to extract the modelName to the header
+
+## Phase 2
+
+### CRD spec
+```golang
+
+type InferenceObjectives struct {
+  metav1.TypeMeta
+  metav1.ObjectMeta
+
+  Spec InferenceObjectivesSpec
+}
+
+type InferenceObjectivesSpec struct {
+	// Scope: Identifier
+    ID string
```

Comment:
Clarification (also based on L201): this is the ID value set in request header which indicates the request is part of the category (tenant/flow/identity) to which the objective applies?